acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/applink/oauth/pageobjects/AuthorizeTokenPage.java

```diff
@@ -0,0 +1,87 @@
+package it.com.atlassian.bitbucket.jenkins.internal.applink.oauth.pageobjects;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.jenkinsci.test.acceptance.junit.Wait;
+import org.jenkinsci.test.acceptance.po.CapybaraPortingLayer;
+import org.jenkinsci.test.acceptance.po.Jenkins;
+import org.jenkinsci.test.acceptance.po.PageObject;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import java.net.URI;
+import java.net.URL;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Optional.ofNullable;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalToIgnoringWhiteSpace;
+
+/**
+ * The page where user authorizes an OAuth request token.
+ */
+public class AuthorizeTokenPage extends PageObject {
+
+    public static final long TIMEOUT_SECONDS = 5L;
+
+    private final String requestToken;
+
+    public AuthorizeTokenPage(Jenkins context, URL pageUrl, String requestTokenValue) {
+        super(context, pageUrl);
+        this.requestToken = requestTokenValue;
+    }
+
+    /**
+     * Authorizes the request token and returns the {@code oauth_verifier} query parameter from the redirect URL
+     *
+     * @return the {@code oauth_verifier} query parameter from the redirect URL after authorizing the token
+     */
+    public String authorize() {
+        openAndVerify();
+        control(By.cssSelector("span[name=\"authorize\"] > span.first-child > button")).click();
```

Comment:
Since we have control over everything, can we simply use id? Selenium best practise says that 
> Preferred selector order : id > name > css > xpath

Reply:
[] change css selector

Reply:
[ ] test task

Reply:
So, this is the html around the submit button:
```
<span class="yui-button yui-submit-button submit-button primary" id="yui-gen1" name="authorize" path="/authorize">
	<span class="first-child">
		<button id="yui-gen1-button" tabindex="0" type="button">Authorize</button>
	</span>
</span>
```
Just so I'm clear, are you suggesting we change the id's to custom ones (if that's possible with jelly)? I'm a bit hesitant to use the yui-generated id's.